ipld/resolver/src/observe.rs

```diff
@@ -0,0 +1,312 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: MIT
+use ipc_observability::{
+    impl_traceable, impl_traceables, register_metrics, Recordable, TraceLevel, Traceable,
+};
+use lazy_static::lazy_static;
+use libipld::cid::Cid;
+use libp2p::gossipsub::TopicHash;
+use libp2p::PeerId;
+use prometheus::{
+    register_histogram, register_int_counter, register_int_gauge, Histogram, IntCounter, IntGauge,
+    Registry,
+};
+use std::error::Error;
+use std::time::Duration;
+
+register_metrics! {
+    PING_RTT: Histogram =
+        register_histogram!("ping_rtt", "Ping roundtrip time");
```

Comment:
I think we need to prefix all of the metrics with `ipld_resolver`. Otherwise just seeing `ping_rtt` metric on it's own does not say much.

Reply:
Fixed: https://github.com/consensus-shipyard/ipc/pull/1207/commits/2037c80cd86f27ac456534f7d1d8da400df81b25